### ISSUE TITLE
Can not create object with nested property set to null in 5.0.9, worked in 4.13.5

### ISSUE THREAD

**Issue by fhnstephen** [Automattic/mongoose#6213]
Node.js version: 6.10.1
Mongoose version: 5.0.9
Mongodb version: 3.4.10

Code to reproduce the bug:
```javascript
mongoose = require('mongoose');
const Bar = {
  enabled: {type: Boolean, default: true},
};

const Foo = new mongoose.Schema({
  nested: {
    bar: Bar,
  },
}, {
  strict: false,
});

mongoose.model('Foo', Foo);

mongoose.connect("mongodb://localhost:27017", () => {
  mongoose.model('Foo').create({nested: null}, function(err, msg) {console.log({err, msg, nested: msg.nested})});
});
```

# Expected Behavior
The creation should work, and the created document should be ```{nested: null}```,  which is the behavior of ```mongoose@4.13.5```.

# Actual Behavior

In mongoose 5.0.9, it outputs the following message to console:

```
{ err:
   TypeError: Cannot read property 'bar' of null
       at $__applyDefaults (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/document.js:273:20)
       at model.Document (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/document.js:102:3)
       at model.Model (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/model.js:60:12)
       at new model (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/model.js:3752:13)
       at toExecute.push.error (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/model.js:2054:22)
       at /Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_async@2.1.4@async/internal/parallel.js:27:9
       at eachOfArrayLike (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_async@2.1.4@async/eachOf.js:57:9)
       at exports.default (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_async@2.1.4@async/eachOf.js:9:5)
       at _parallel (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_async@2.1.4@async/internal/parallel.js:26:5)
       at parallelLimit (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_async@2.1.4@async/parallel.js:85:26)
       at utils.promiseOrCallback.cb (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/model.js:2064:5)
       at Object.promiseOrCallback (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/utils.js:211:14)
       at Function.create (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/model.js:2029:16)
       at mongoose.connect (/Users/fhnstephen/github/clarity/ClarityCloud/src/api/node_api/test.js:18:25)
       at promise.then (/Users/fhnstephen/github/clarity/ClarityCloud/node_modules/_mongoose@5.0.9@mongoose/lib/connection.js:419:24)
       at process._tickCallback (internal/process/next_tick.js:109:7),
  msg: [],
  nested: undefined }
```

After some trials, I found that if I remove `strict: false`, the error disappeared. But that is not a solution to my project.

Am I writing a wrong schema? Or is it a bug in `mongoose@5`?

**Comment by lineus** [comment]
this is related to #6187 . I applied the [patch](https://github.com/Automattic/mongoose/commit/ab142870f39403608406ee64a7a35ce844fc2935#diff-fc7fc7567205f95715daa4204b8ea0b0) that addresses #6187 and it works. The fix will be deployed in 5.0.10

### Code used to duplicate:
```
#!/usr/bin/env node
'use strict'

const mongoose = require('./lib/test_db')
const Schema = mongoose.Schema

const Bar = {
  enabled: {
    type: Boolean,
    default: true
  }
}

const Foo = new Schema({
  nested: {
    bar: Bar
  }
}, {
  strict: false
})

mongoose.model('Foo', Foo)

mongoose.model('Foo').create({
  nested: null
}, function (err, msg) {
  mongoose.connection.close()
  if (err) { return console.error(err) }
  console.log(msg)
})
```
### output after patch:
```
InspiredMacPro:mongoose5 lineus$ ./6213.js 
{ nested: null, _id: 5a9fe0d6bae8649ba5d10664, __v: 0 }
InspiredMacPro:mongoose5 lineus$ 
```

**Comment by vkarpov15** [comment]
@lineus is right, this looks like a duplicate of #6187. 5.0.10 will be released shortly with a fix :+1:
